### PR TITLE
implemented authorS to match multiple authors from archives

### LINKED ISSUE #249: Wrong authors for imported articles
Sometimes, we attribute the wrong author to a given article (translation?) when importing old data. It has to be discussed with editors to find pathologic cases.

### PR DESCRIPTION
**Pull request by vinyll**
Fix #249.
Backend implementation for #258.

### CODE REVIEW

File: cafebabel/articles/models.py
Comment (davidbgk) [inline cafebabel/articles/models.py]: Is the `len()` necessary?

File: cafebabel/articles/models.py
Comment (davidbgk) [inline cafebabel/articles/models.py]: A comment/todo would be cool to understand why we are doing this for now.

File: cafebabel/articles/translations/views.py
Comment (davidbgk) [inline cafebabel/articles/translations/views.py]: Why not `article.authors`?

File: cafebabel/articles/models.py
Comment (davidbgk) [inline cafebabel/articles/models.py]: I'm afraid it erases previous multiple authors?

File: cafebabel/django_migrations.py
Comment (davidbgk) [inline cafebabel/django_migrations.py]: Why not `original_article.authors`?

**Review by davidbgk — CHANGES_REQUESTED** [review]
I think there are still missing tests for edge cases.

File: cafebabel/articles/translations/views.py
Comment (vinyll) [reply cafebabel/articles/translations/views.py]: That makes me wonder what should happen when updating the list of the original article's authors.
I suppose the translations should follow.

**Comment by vinyll** [comment]
> I think there are still missing tests for edge cases.

I guess at a certain point we should decide to implement the functionality on the frontend side and this should avoid most edge-cases.

Seeing the management of the authors for translations makes me wonder if Translation authors shouldn't actually be a virtual property instead reading from `original_article.authors`?

**Review by vinyll — COMMENTED** [review]
LGTM

**Comment by davidbgk** [comment]
> LGTM

Still importing locally, I'll not approve before ;)

**Comment by davidbgk** [comment]
For the moment, authors for articles are fixed with that PR but not translators which remains the first author (which appears to be wrong)

**Comment by vinyll** [comment]
Indeed, the translations have a `.translators()` methods just like the articles have `.authors()`.
That will need to be implemented…

File: cafebabel/tests/test_users.py
Comment (vinyll) [inline cafebabel/tests/test_users.py:123]: We shall need to be consistent between `pk` and `id` though the project. We've been mostly using `id`, and seldomly `pk`.

**Comment by vinyll** [comment]
Tests are fixed, @cafebabel-webmaster may want to customize what has been adapted on the frontend.
Appart of that it seems to be all correct…

**Comment by johangiraud** [comment]
Ok I'll work on it now

File: cafebabel/static/css/styles.css
Comment (davidbgk) [inline cafebabel/static/css/styles.css]: I think these styles could be in their dedicated custom CSS, this way…

File: cafebabel/templates/articles/edit.html
Comment (davidbgk) [inline cafebabel/templates/articles/edit.html]: …it will be possible to put them here after the lib one and…

File: cafebabel/templates/base.html
Comment (davidbgk) [inline cafebabel/templates/base.html]: …to re-swap this block below the `style.css` inclusion.

File: cafebabel/static/css/styles.css
Comment (johangiraud) [reply cafebabel/static/css/styles.css]: but we can't overwrite their css within their css files, right ? so ?

Comment (davidbgk) [reply cafebabel/static/css/styles.css]: No, but you can create a new dedicated CSS with custom styles for that component.

Comment (johangiraud) [reply cafebabel/static/css/styles.css]: 👍

File: cafebabel/templates/articles/edit.html
Comment (vinyll) [inline cafebabel/templates/articles/edit.html]: you may want to give it a more generic class name when it will be reused. Something like "select-multi" or whatever.

File: cafebabel/templates/base.html
Comment (johangiraud) [reply cafebabel/templates/base.html]: ok ! got it !

File: cafebabel/templates/articles/macros.html
Comment (vinyll) [inline cafebabel/templates/articles/macros.html]: unindent this block

Comment (johangiraud) [reply cafebabel/templates/articles/macros.html]: ok https://github.com/cafebabel/cafebabel.com/pull/260/commits/562cb827bbaefa2fa0e9f4e865cee07f17724622

File: cafebabel/templates/articles/edit.html
Comment (johangiraud) [reply cafebabel/templates/articles/edit.html]: ok https://github.com/cafebabel/cafebabel.com/pull/260/commits/6c8b5fea95ac0f78636f41bc0599309d55ee5056

**Comment by johangiraud** [comment]
in _test_translation.py_ I killed:
```
def test_translation_published_should_have_translator(client, translation):
    translation.status = 'published'
    translation.save()
    translator = translation.translators[0]
    response = client.get(f'/en/article/{translation.slug}-{translation.id}/')
    assert response.status_code == HTTPStatus.OK
    assert ((f'Translated from '
             f'<a href="/en/article/draft/{translation.original_article.id}/">'
             f'article title') in response)
    assert (f'by <a href="/en/profile/{translator.id}/">{translator}</a>.'
            in response)
```
because I now display the translator right next to the author(s). And I no longer display it at the article bottom. I don't get how to change properly the test. Can you help ?

File: cafebabel/tests/test_translations.py
Comment (vinyll) [inline cafebabel/tests/test_translations.py]: That's light by I get the workaround

Comment (johangiraud) [reply cafebabel/tests/test_translations.py]: 👍

**Comment by johangiraud** [comment]
capture d ecran 2018-03-07 a 16 55 37
ok here is the problem. I have juste created a translation and I land on this page. I would like to see displayed the translator profile link right next to the author(s) like that:
capture d ecran 2018-03-07 a 16 59 01
How do I do that?
I tried `{% if article.is_translation or article.is_translations_drafts %}` in _article/macros.html_ but it doesn't work..

**Comment by davidbgk** [comment]
>  How do I do that?

@cafebabel-webmaster it's because we do not call the `article_meta` macro in `translations/detail.html` template, 2 options: either use it or adapt directly the template.

**Comment by johangiraud** [comment]
> @cafebabel-webmaster it's because we do not call the article_meta macro in translations/detail.html template, 2 options: either use it or adapt directly the template.

We do line 3 : `{% import "articles/macros.html" as macros without context %}` @davidbgk 

I can't have it work :-( `{% if article.is_translation or article.is_translation %}`?

**Review by davidbgk — APPROVED** [review]
🚀

File: cafebabel/templates/articles/to-translate.html
Comment (vinyll) [inline cafebabel/templates/articles/to-translate.html]: `without context` is implicit; can be removed

**Pull request merged**
